### PR TITLE
PE-462: download drive

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: src/commands/download_drive.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { cliArDriveAnonymousFactory, cliArDriveFactory } from '../index';
+import { CLICommand, ParametersHelper } from '../CLICommand';
+import { CLIAction } from '../CLICommand/action';
+import { DrivePrivacyParameters, DriveIdParameter, LocalPathParameter } from '../parameter_declarations';
+import { getOutputFolderPathAndName } from '../utils';
+import { join as joinPath } from 'path';
+
+new CLICommand({
+	name: 'download-drive',
+	parameters: [DriveIdParameter, { name: LocalPathParameter, description: 'TODO' }, ...DrivePrivacyParameters],
```
Comment (arielmelendez) [inline src/commands/download_drive.ts:11]: Description unfinished

File: src/commands/download_drive.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		const maxDepth = await parameters.getMaxDepth(Number.MAX_SAFE_INTEGER);
+		const destOutputPath = parameters.getParameterValue(LocalPathParameter) || '.';
+		const [destFolderPath, customFolderName] = getOutputFolderPathAndName(destOutputPath);
+		let outputPath: string;
+
+		if (await parameters.getIsPrivate()) {
+			const wallet = await parameters.getRequiredWallet();
+			const ardrive = cliArDriveFactory({
+				wallet,
+				feeMultiple: parameters.getOptionalBoostSetting()
```
Comment (arielmelendez) [inline src/commands/download_drive.ts:24]: You can omit this since we're just reading.

File: src/commands/download_drive.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { join as joinPath } from 'path';
+
+new CLICommand({
+	name: 'download-drive',
+	parameters: [DriveIdParameter, { name: LocalPathParameter, description: 'TODO' }, ...DrivePrivacyParameters],
+	action: new CLIAction(async (options) => {
+		const parameters = new ParametersHelper(options);
+		const driveId = parameters.getRequiredParameterValue(DriveIdParameter, EID);
+		const maxDepth = await parameters.getMaxDepth(Number.MAX_SAFE_INTEGER);
+		const destOutputPath = parameters.getParameterValue(LocalPathParameter) || '.';
```
Comment (arielmelendez) [inline src/commands/download_drive.ts:16]: Won't this have the wrong description text?

**Review by arielmelendez — CHANGES_REQUESTED** [review]
A few issues

File: src/commands/download_drive.ts (hunk 1 above)
Comment (matibat) [reply src/commands/download_drive.ts]: Updated 4f6700e5d4eb8239d30f6f0897ebcb34a5c3182e

File: src/commands/download_drive.ts (hunk 3 above)
Comment (matibat) [reply src/commands/download_drive.ts:24]: If it was regarding the parameter description, then: 4f6700e5d4eb8239d30f6f0897ebcb34a5c3182e

File: src/commands/download_drive.ts (hunk 2 above)
Comment (matibat) [reply src/commands/download_drive.ts]: Nice catch! cd48e8a37b1acb67eaea67c4c0a377258cf5a7cd

**Review by arielmelendez — DISMISSED** [review]

File: src/commands/index.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ import './create_folder';
 import './create_manifest';
 import './create_tx';
 import './download_file';
+import './download_drive';
```
Comment (arielmelendez) [inline src/commands/index.ts:8]: This is not in alphabetical order

File: README.md [hunk 5]
```diff
@@ -717,9 +718,19 @@ ardrive download-file -w /path/to/wallet.json -file-id "ff450770-a9cb-46a5-9234-
...
 
+### Downloading a Drive
+
+To download the whole drive you can use the `download-drive` command.
+
+```shell
+ardrive download-drive -d "c0c8ba1c-efc5-420d-a07c-a755dc67f6b2"
+```
+
+This is equivalent to run the `download-folder` command against the root folder of the drive. So the same behavior is expected.
```
Comment (arielmelendez) [inline README.md:732]:
```suggestion
This is equivalent to running the `download-folder` command against the root folder of the drive.
```
Grammar correction and removing redundant sentence.

File: src/commands/index.ts (hunk 4 above)
Comment (matibat) [reply src/commands/index.ts:7]: f6ba56f

File: src/parameter_declarations.ts [hunk 6]
```diff
@@ -242,17 +241,6 @@ Parameter.declare({
 Parameter.declare({
 	name: LocalPathParameter,
 	aliases: ['--local-path'],
-	description: `the path on the local filesystem for the file that will be uploaded
```
Comment (arielmelendez) [inline src/parameter_declarations.ts:245]: Isn't this going to destroy the documentation for `upload-file`'s --local-path use case?

Comment (matibat) [reply src/parameter_declarations.ts]: This should fix that. 744a5e097b4d247abd5d218e2a8210ecb048b7b3

**Review by arielmelendez — APPROVED** [review]

**Review by fedellen — APPROVED** [review]

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**